### ISSUE TITLE
Editing a case contact should not change who the case contact was created by

### ISSUE THREAD

**Issue by Zrrrpy** [rubyforgood/casa#426]
Part of epics #59 + #4 and #105

**What type of user is this for? [admin/volunteer/supervisor/all]**
admins and supervisors, but affects volunteers

**Where does/should this occur?**
Triggered by any user editing any case contact not created by themselves. Relates to the `creator_id` associated with existing `case_contacts` in `create_table "case contacts"` 

**Description**
When any user edits a case contact not created by themselves, the original `creator_id` should remain intact and not be overridden by the user making the edit. 

**Relates to issues:**
#408, #393

**Comment by GuillermoCoding** [comment]
I was starting to dig into this issue and had some questions.

**Comment by GuillermoCoding** [comment]
I was starting to dig into this issue and had some questions.
When logged in as an admin user,  I tried to submit an edit an existing case contact and I was getting this error when testing locally.

<img width="500" alt="image" src="https://user-images.githubusercontent.com/18248767/88446840-6ef11480-cdfb-11ea-9a7f-ba958c3efd55.png">

I was looking into why I was getting this error when updating a case contact and looks like it was coming from `case_contacts_controller.rb` since `current_user` (admin) did not have any casa cases
<img width="706" alt="image" src="https://user-images.githubusercontent.com/18248767/88446889-c55e5300-cdfb-11ea-9445-895ca7613166.png">

I changed `current_user.casa_cases.first` to `@contact_case.casa_case` and I was able to edit the case contact as an admin.

I was wondering if I should also make this change to allow admins to make edits or was the error that I was getting expected behavior?
